cache/remotecache/s3/s3.go

```diff
@@ -464,6 +472,16 @@ func (s3Client *s3Client) ReaderAt(ctx context.Context, desc ocispecs.Descriptor
 	return &readerAt{ReaderAtCloser: readerAtCloser, size: desc.Size}, nil
 }
 
+func (s3Client *s3Client) Info(ctx context.Context, dgst digest.Digest) (content.Info, error) {
+	key := s3Client.blobKey(dgst)
+	head, err := s3Client.head(ctx, key)
+	if err != nil || head.ContentLength == nil {
+		return content.Info{}, err
+	}
+
+	return content.Info{Digest: dgst, Size: *head.ContentLength}, nil
```

Comment:
Missing other fields.

Reply:
We should avoid the HTTP request for this call. Instead wrap `desc` to a s3Client in `makeDescriptorProviderPair` so the client in here has access to the descriptor.